### PR TITLE
fix(typeahead): input was not always updated when model changes

### LINKED ISSUE #2816: it is not possible to reset input value for typeahead component
### Bug description:

For the typeahead component, it is not possible to clear the model after the results list appears. If the results list doesn't appear, the model can be cleared from a button without problem

### Steps to reproduce:
- open [https://stackblitz.com/edit/angular-rybwwn-w3w4ca](https://stackblitz.com/edit/angular-rybwwn-w3w4ca)
- introduce text "aaa"
- click on button "Clear"
- the input is cleared
- introduce text "ala"
- click on button "Clear"
- the input is not cleared

You can fork a StackBlitz from one of our [demos](https://ng-bootstrap.github.io/#/components) and use it as a starting point.

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 6.1.0

ng-bootstrap: 3.3.0

Bootstrap: 4.0

### PR DESCRIPTION
**Pull request by fbasso**
fix #2816

### CODE REVIEW

**Comment by benouat** [comment]
Could you please change your commit message and PR title to be fix(typeahead) ?

File: src/typeahead/typeahead.spec.ts [hunk 1]
```diff
@@ -462,6 +462,31 @@ describe('ngb-typeahead', () => {
          tick(250);
          expectWindowResults(compiled, ['+one', 'one more']);
        }));
+    it('should stick model on click outside', async(() => {
```
Comment (maxokorokov) [inline src/typeahead/typeahead.spec.ts]: Sorry, I don't understand the message

File: src/typeahead/typeahead.ts [hunk 2]
```diff
@@ -271,6 +270,7 @@ export class NgbTypeahead implements ControlValueAccessor,
           event.preventDefault();
           this._resubscribeTypeahead.next(null);
           this.dismissPopup();
+          this._writeInputValue(this._inputValueBackup);
```
Comment (maxokorokov) [inline src/typeahead/typeahead.ts]: I don't get the fix, why do it only on ESC and not on document click for instance?
Also I think `dismissPopup` can be called manually from code, will it work?

File: src/typeahead/typeahead.ts [hunk 3]
```diff
@@ -224,7 +224,6 @@ export class NgbTypeahead implements ControlValueAccessor,
   dismissPopup() {
     if (this.isPopupOpen()) {
       this._closePopup();
-      this._writeInputValue(this._inputValueBackup);
```
Comment (maxokorokov) [inline src/typeahead/typeahead.ts]: As this is `showHint`-related, maybe just wrap this in:

```ts
if (this.showHint) {
  this._writeInputValue(this._inputValueBackup);
}
```

**Comment by benouat** [comment]
After discussion with @maxokorokov and @fbasso I will work today on a different fix for that underlying issue #2816.

I will close that PR as soon as I open the new one.

**Pull request closed without merging**